### PR TITLE
Automatically port some remarks for System.Security.Cryptography (.NET Core 2.X)

### PR DESCRIPTION
**Pull request by carlossanlop**
Total modified files: 3
    - D:\dotnet-api-docs\xml\System.Security.Cryptography\RSAEncryptionPaddingMode.xml
    - D:\dotnet-api-docs\xml\System.Security.Cryptography\RSASignaturePaddingMode.xml
    - D:\dotnet-api-docs\xml\System.Security.Cryptography\HashAlgorithmName.xml

Total modified assemblies: 3
    - System.Security.Cryptography.RSAEncryptionPaddingMode
    - System.Security.Cryptography.RSASignaturePaddingMode
    - System.Security.Cryptography.HashAlgorithmName

Total modified containers: 0

Total modified APIs: 5
    - F:System.Security.Cryptography.RSAEncryptionPaddingMode.Pkcs1
    - F:System.Security.Cryptography.RSAEncryptionPaddingMode.Oaep
    - F:System.Security.Cryptography.RSASignaturePaddingMode.Pkcs1
    - F:System.Security.Cryptography.RSASignaturePaddingMode.Pss
    - P:System.Security.Cryptography.HashAlgorithmName.Name

Total problematic APIs: 0

Total added exceptions: 0

Total modified individual elements: 5

### CODE REVIEW

File: xml/System.Security.Cryptography/HashAlgorithmName.xml [hunk 1]
```diff
@@ -282,7 +282,15 @@ Due to collision problems with MD5 and SHA1, Microsoft recommends a security mod
       <Docs>
         <summary>Gets the underlying string representation of the algorithm name.</summary>
         <value>The string representation of the algorithm name, or <see langword="null" /> or <see cref="F:System.String.Empty" /> if no hash algorithm is available.</value>
-        <remarks>To be added.</remarks>
+        <remarks>
+          <format type="text/markdown"><![CDATA[
+
+## Remarks
+
+May be `null` or empty to indicate that no hash algorithm is applicable.
```
Comment (bartonjs) [inline xml/System.Security.Cryptography/HashAlgorithmName.xml]: `null` and empty have no special defined meaning on this type.  It's simply a struct, and therefore can have been initialized to default.

Remove this remark.

File: xml/System.Security.Cryptography/RSASignaturePaddingMode.xml [hunk 2]
```diff
@@ -107,6 +116,15 @@
       <MemberValue>1</MemberValue>
       <Docs>
         <summary>Probabilistic Signature Scheme</summary>
+        <remarks>
+          <format type="text/markdown"><![CDATA[
+
+## Remarks
+
+This corresponds to the RSASSA-PKCS1-v1.5 signature scheme of the PKCS #1 RSA Encryption Standard. It is recommended for new applications.
```
Comment (bartonjs) [inline xml/System.Security.Cryptography/RSASignaturePaddingMode.xml]:
```suggestion
This corresponds to the RSASSA-PSS signature scheme of the PKCS #1 RSA Encryption Standard. It is recommended for new applications.
```

File: xml/System.Security.Cryptography/HashAlgorithmName.xml (hunk 1 above)
Comment (rpetrusha) [reply xml/System.Security.Cryptography/HashAlgorithmName.xml]: This is just a repetition of the \<value> section, where it is better placed. This should be deleted, and the \<value> section modified to reflect the comment by @bartonjs.

File: xml/System.Security.Cryptography/RSAEncryptionPaddingMode.xml [hunk 3]
```diff
@@ -71,6 +71,15 @@ The `Oaep` mode corresponds to the RSAES-OEAP encryption scheme described in the
       <MemberValue>1</MemberValue>
       <Docs>
         <summary>Optimal Asymmetric Encryption Padding. It is recommended for new applications.</summary>
+        <remarks>
```
Comment (rpetrusha) [inline xml/System.Security.Cryptography/RSAEncryptionPaddingMode.xml]: This should be deleted. It appears in the \<MemberGroup> section, and individual enum \<remarks> sections are suppressed from the build.

File: xml/System.Security.Cryptography/RSAEncryptionPaddingMode.xml [hunk 4]
```diff
@@ -105,6 +114,15 @@ The `Oaep` mode corresponds to the RSAES-OEAP encryption scheme described in the
       <MemberValue>0</MemberValue>
       <Docs>
         <summary>PKCS #1 v1.5. It is supported for compatibility with existing applications.</summary>
+        <remarks>
```
Comment (rpetrusha) [inline xml/System.Security.Cryptography/RSAEncryptionPaddingMode.xml]: Similarly, this should be deleted.

File: xml/System.Security.Cryptography/RSASignaturePaddingMode.xml [hunk 5]
```diff
@@ -73,6 +73,15 @@
       <MemberValue>0</MemberValue>
       <Docs>
         <summary>PKCS #1 v1.5</summary>
+        <remarks>
```
Comment (rpetrusha) [inline xml/System.Security.Cryptography/RSASignaturePaddingMode.xml]: This should be deleted. See above.

File: xml/System.Security.Cryptography/RSASignaturePaddingMode.xml [hunk 6]
```diff
@@ -107,6 +116,15 @@
       <MemberValue>1</MemberValue>
       <Docs>
         <summary>Probabilistic Signature Scheme</summary>
+        <remarks>
```
Comment (rpetrusha) [inline xml/System.Security.Cryptography/RSASignaturePaddingMode.xml]: This should be deleted; see above. The PSS correction is already present in the text.

File: xml/System.Security.Cryptography/RSAEncryptionPaddingMode.xml (hunk 3 above)
Comment (carlossanlop) [reply xml/System.Security.Cryptography/RSAEncryptionPaddingMode.xml]: I don't see a MemberGroup, but I think you meant the general remark for the whole Type, @rpetrusha. I'll remove it.

File: xml/System.Security.Cryptography/HashAlgorithmName.xml [hunk 7]
```diff
@@ -281,7 +281,7 @@ Due to collision problems with MD5 and SHA1, Microsoft recommends a security mod
       </ReturnValue>
       <Docs>
         <summary>Gets the underlying string representation of the algorithm name.</summary>
-        <value>The string representation of the algorithm name, or <see langword="null" /> or <see cref="F:System.String.Empty" /> if no hash algorithm is available.</value>
+        <value>The string representation of the algorithm name, or <see langword="default" /> if no hash algorithm is available.</value>
```
Comment (carlossanlop) [inline xml/System.Security.Cryptography/HashAlgorithmName.xml]: @bartonjs I addressed your comment about null and empty, by removing the remark and (as suggested by @rpetrusha) modifying the value: substituted null and empty with default. Please confirm this was the correct change.

File: xml/System.Security.Cryptography/RSASignaturePaddingMode.xml [hunk 8]
```diff
@@ -35,9 +35,9 @@
 
 `Pkcs1` corresponds to the RSASSA-PKCS1-v1.5 signature scheme of the [PKCS #1: RSA Encryption Standard](http://www.emc.com/emc-plus/rsa-labs/standards-initiatives/pkcs-rsa-cryptography-standard.htm). It is supported for compatibility with existing applications.  
 
-`Pss` corresponds to the Probabilistic Signature Scheme (PSS) of the [PKCS #1: RSA Encryption Standard](http://www.emc.com/emc-plus/rsa-labs/standards-initiatives/pkcs-rsa-cryptography-standard.htm). It is recommended for new applications.
+`Pss` corresponds to the Probabilistic Signature Scheme (RSASSA-PSS) of the [PKCS #1: RSA Encryption Standard](http://www.emc.com/emc-plus/rsa-labs/standards-initiatives/pkcs-rsa-cryptography-standard.htm). It is recommended for new applications.
```
Comment (carlossanlop) [inline xml/System.Security.Cryptography/RSASignaturePaddingMode.xml]: @bartonjs I addressed the suggestion you originally gave me in the remark for the PPS enum value by adding your acronym to the general description found in the enum Type. Please let me know if this was the correct change to make.

File: xml/System.Security.Cryptography/HashAlgorithmName.xml (hunk 7 above)
Comment (bartonjs) [reply xml/System.Security.Cryptography/HashAlgorithmName.xml]: I'd just leave it as it is entirely, or delete everything in the "or" clause.

Like all null values it's "do nothing", "do the default thing", or "be invalid", determined by the caller.

File: xml/System.Security.Cryptography/RSASignaturePaddingMode.xml (hunk 8 above)
Comment (bartonjs) [reply xml/System.Security.Cryptography/RSASignaturePaddingMode.xml]: "Probabilistic Signature Scheme" is "PSS".  RSA-signature-using-PSS is "RSASSA-PSS" (RSA Signature Scheme with Appendix  - Probabilistic Signature Scheme).

So where this is used, it was correct as-is, and is incorrect with the change.

If we want to include RSASSA-PSS in the string, it'd be something like

```suggestion
`Pss` corresponds to the RSASSA-PSS signature scheme of the [PKCS #1: RSA Encryption Standard](http://www.emc.com/emc-plus/rsa-labs/standards-initiatives/pkcs-rsa-cryptography-standard.htm). It is recommended for new applications.
```

I'd just leave it as-is.

**Comment by rpetrusha** [comment]
I'll merge this now, @carlossanlop.

**Pull request merged**